src/main/java/com/segment/analytics/android/integrations/comscore/ComScoreIntegration.java

```diff
@@ -450,10 +455,63 @@ private AdvertisementMetadata getAdvertisementMetadata(Map<String, String> mappe
     return new AdvertisementMetadata.Builder().customLabels(mappedAdProperties).build();
   }
 
+  public HashMap<String, String> setConsentLabelValue(Map<String, String> main, Map<String, String> fallback, Settings settings) {
+    if (settings.getConsentFlagProp() != null && !settings.getConsentFlagProp().trim().isEmpty() ) {
+      String consentFlagKey = settings.getConsentFlagProp();
+      String consentFlagValue;
+      if (main != null && main.containsKey(consentFlagKey)) {
+        consentFlagValue = main.get(consentFlagKey);
+      } else if (fallback != null && fallback.containsKey(consentFlagKey) ) {
+        consentFlagValue = fallback.get(consentFlagKey);
+      } else {
+        consentFlagValue = null;
+      }
+
+      if (consentFlagValue != null)
+      {
+        // Parse consent flag value
+        Pattern privacyStringPattern = Pattern.compile("^1(-|Y|N){3}");
```

Comment:
since this is constant, maybe we should initialize it outside once rather than everytime we call this function

Reply:
Done, moved the privacyStringPattern outside the function.